docs/developers/hacking.rst

```diff
@@ -188,6 +188,156 @@ virtual environment (using the :command:`workon` command) and running the
 Happy hacking!!
 
 
+.. _hacking#windows:
+
+Setting up the development environment on Windows
+-------------------------------------------------
+
+.. note:: Ensure that you are executing all of the following steps with Administrator privileges!
+
+Install Redis
+^^^^^^^^^^^^^
+
+Download the latest installer from:
+    https://github.com/MSOpenTech/redis/releases
+
+During the installation you will be asked to set what port Redis should listen on; leave it at the default (6379).
+
+Install Nodejs
+^^^^^^^^^^^^^^
+
+Download the latest installer from:
+    https://nodejs.org/en/
+
+Pootle Set Up
+^^^^^^^^^^^^^
+
+.. note:: For convenient these instructions consistently specify paths ``C:\venv`` and ``C:\git\pootle``, but you can change these to suit your environment and needs.
+
+For installing the dependencies in an isolated environment, set up a fresh virtualenv.
+
+.. code-block:: console
+
+    > python -m pip install virtualenv
+    > virtualenv C:\venv
+
+Activate the new virtualenv and upgrade pip:
+
+.. code-block:: console
+
+    > C:\venv\Scripts\activate
+    (venv)> python -m pip install --upgrade pip
+
+Go to the pootle ``requirements\base.txt`` and comment out the following packages:
+
+.. code-block:: console
+
+    # lxml
+    # levenshtein
+    # scandir
+
+These three packages are difficult to build on Windows, so we will download pre-built versions to install manually 
+further on.
+
+Also change any ``~=`` compatible-version markers into ``==`` exact-version markers, 
+as Windows-based ``pip`` doesn't seem to like these.
+
+To avoid permissions issues, required packages have to be downloaded and installed in separate steps. 
+By default, pip stores temporary files in your ``user\AppData`` folder which may not allow access in 
+all circumstances. There are also some packages that need to be downloaded and installed manually.
+Create a temporary folder somewhere that any user can access without permission issues (e.g. ``C:\temp``).
+If you are seeing "directory was not empty" or "file not found" errors during a ``pip install``, then these 
```

Comment:
I'd put these following commands in a `notes` block so that they're not part of the normal flow.